flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/model/StartingPosition.java

```diff
@@ -0,0 +1,106 @@
+/*
```

Comment:
This class looks quite like the `SentinelSequenceNumber ` class, is there any possibility to combine those two classes?

Reply:
We cannot use `SentinelSequenceNumber` to achieve what this class does, as `SentinelSequenceNumber` is an enum. We could get rid of `SentinelSequenceNumber` in favour of this, however that class has a lot of references and would be a big refactor. Also I think semantically they differ.

Also looking at the code, `SentinelSequenceNumber` is potentially serialised and therefore we would need to migrate snapshots or handle deserialisation mappings which is risky.